xla/client/xla_builder_test.cc

```diff
@@ -418,6 +418,22 @@ TEST_F(XlaBuilderTest, AllGatherR2) {
       ShapeUtil::Equal(root->shape(), ShapeUtil::MakeShape(F32, {4, 64})));
 }
 
+TEST_F(XlaBuilderTest, AllGatherWithTuple) {
+  XlaBuilder b(TestName());
+  auto x = Parameter(&b, 0, ShapeUtil::MakeShape(F32, {4}), "x");
+  auto x2 = Parameter(&b, 1, ShapeUtil::MakeShape(F32, {16, 4}), "x2");
+  AllGather(Tuple(&b, {x, x2}), /*all_gather_dimension=*/0,
```

Comment:
From an API perspective, XlaBuilder has:

```
  XlaOp AllToAll(XlaOp operand, int64_t split_dimension,
                 int64_t concat_dimension, int64_t split_count,
                 absl::Span<const ReplicaGroup> replica_groups,
                 const std::optional<Layout>& layout = std::nullopt,
                 const std::optional<ChannelHandle>& channel_id = std::nullopt);

  XlaOp AllToAllTuple(
      absl::Span<const XlaOp> operands,
      absl::Span<const ReplicaGroup> replica_groups,
      const std::optional<Layout>& layout,
      const std::optional<ChannelHandle>& channel_id = std::nullopt);

XlaOp AllReduce(XlaOp operand, const XlaComputation& computation,
                absl::Span<const ReplicaGroup> replica_groups = {},
                const std::optional<ChannelHandle>& channel_id = std::nullopt,
                const std::optional<Shape>& shape_with_layout = std::nullopt,
                std::optional<bool> use_global_device_ids = std::nullopt);

XlaOp AllReduceTuple(
    absl::Span<const XlaOp> operand, const XlaComputation& computation,
    absl::Span<const ReplicaGroup> replica_groups = {},
    const std::optional<ChannelHandle>& channel_id = std::nullopt,
    const std::optional<Shape>& shape_with_layout = std::nullopt,
    std::optional<bool> use_global_device_ids = std::nullopt);
```

Consider adding an `AllGatherTuple` API for consistency, and using it in this test. See AllReduceTuple impl, it just wraps in tuple and dispatches.

Reply:
Thanks. I think it makes sense to make it consistent. I will take a look at this.